src/autogluon/bench/eval/evaluation/preprocess/preprocess_openml.py

```diff
@@ -137,40 +138,35 @@ def preprocess_openml_input(
 
     cleaned_input = preprocess_utils.clean_result(raw_input, folds_to_keep=folds_to_keep, remove_invalid=False)
 
-    cleaned_input["tid"] = [
-        int(part) for x in cleaned_input["id"] for part in re.split(r"[/_]", x)[-1:] if part.isdigit()
-    ]
+    if "tid" not in cleaned_input:
+        if "id" in cleaned_input:
+            cleaned_input["tid"] = [
+                int(part) for x in cleaned_input["id"] for part in re.split(r"[/_]", x)[-1:] if part.isdigit()
+            ]
 
-    """
-    Update tid to the new ones in case the runs are old
-
-    Name                | oldtid | newtid
-
-    KDDCup09-Upselling  | 360115 | 360975
-    MIP-2016-regression | 359947 | 360945
-    QSAR-TID-10980      |  14097 | 360933
-    QSAR-TID-11         |  13854 | 360932
-    """
-    cleaned_input["tid"] = (
-        cleaned_input["tid"]
-        .map(
-            {
-                360115: 360975,
-                359947: 360945,
-                14097: 360933,
-                13854: 360932,
-            }
+    if "tid" in cleaned_input:
+        """
+        Update tid to the new ones in case the runs are old
+
+        Name                | oldtid | newtid
+
+        KDDCup09-Upselling  | 360115 | 360975
+        MIP-2016-regression | 359947 | 360945
+        QSAR-TID-10980      |  14097 | 360933
+        QSAR-TID-11         |  13854 | 360932
+        """
+        cleaned_input["tid"] = (
+            cleaned_input["tid"]
+            .map(
+                {
+                    360115: 360975,
+                    359947: 360945,
+                    14097: 360933,
+                    13854: 360932,
+                }
+            )
+            .fillna(cleaned_input["tid"])
```

Comment:
Is this for specific datasets only?

Reply:
Yes, this is due to 4 datasets having their `tid` updated in OpenML, so we can compare results before and after the update.